var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -629,7 +629,11 @@ def configure_args(self):
         # for versions older than 3.0.3,3.1.3,4.0.0
         # Presumably future versions after 11/2018 should support slurm+static
         if spec.satisfies('schedulers=slurm'):
-            config_args.append('--with-pmi={0}'.format(spec['slurm'].prefix))
+            if spec['slurm'].prefix == '/usr':
+                # slurm is native
+                config_args.append('--with-pmi')
+            else:
```

Comment:
Is it just `/usr` that gives issues? or also e.g. `/opt`? I am wondering if this change is too specific and ought to be generalized a bit.

Reply:
According to my understanding, if the package is installed via apt, the prefix of the slurm package in spack is set to `/usr`.

Reply:
@laekov This is true, but external packages are not solely installed by `apt`.

Reply:
But I guess if the external package is installed via customized compilation, the prefix should work well.